src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -967,12 +967,6 @@ CTranslatorExprToDXL::AddBitmapFilterColumns(
 	// compute what additional columns are required in the output of the (Dynamic) Bitmap Table Scan
 	CColRefSet *pcrsAdditional = GPOS_NEW(mp) CColRefSet(mp);
 
-	if (NULL != pexprRecheckCond)
-	{
-		// add the columns used in the recheck condition
-		pcrsAdditional->Include(pexprRecheckCond->DeriveUsedColumns());
-	}
-
```

Comment:
Excluding attributes that are from recheck condition and at the same time retaining those ones from filter is not solution, in general. We might find an example exposing redundant attributes to the output list of BitmapHeapScan that are member of filter condition but not ones in target list, e.g. the following:
```sql
create table material_bitmapscan_1(i int, j int, k timestamp, l timestamp) with(appendonly=true) distributed replicated;
create index material_bitmapscan_1_idx on material_bitmapscan_1 using btree(k);
insert into material_bitmapscan_1
select i, mod(i, 10), timestamp '2021-06-01' + interval '1' day * mod(i, 30), timestamp '2021-06-01' + interval '1' day * mod(i, 30)
from generate_series(1, 10000) i;

explain (costs off, verbose) with mat as(
    select i, j from material_bitmapscan_1
    where i = 2 and j = 2 and k = timestamp '2021-06-03' and l = timestamp '2021-06-03'
)
select m1.i
from mat m1 join mat m2 on m1.j = m2.j;
                                                                                       QUERY PLAN                                                                                       
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Gather Motion 1:1  (slice1; segments: 1)
   Output: share0_ref3.i
   ->  Sequence
         Output: share0_ref3.i
         ->  Shared Scan (share slice:id 1:0)
               Output: share0_ref1.i, share0_ref1.j
               ->  Materialize
                     Output: material_bitmapscan_1.i, material_bitmapscan_1.j, material_bitmapscan_1.l
                     ->  Bitmap Heap Scan on public.material_bitmapscan_1
                           Output: material_bitmapscan_1.i, material_bitmapscan_1.j, material_bitmapscan_1.l
                           Recheck Cond: (material_bitmapscan_1.k = '2021-06-03 00:00:00'::timestamp without time zone)
                           Filter: ((material_bitmapscan_1.i = 2) AND (material_bitmapscan_1.j = 2) AND (material_bitmapscan_1.l = '2021-06-03 00:00:00'::timestamp without time zone))
                           ->  Bitmap Index Scan on material_bitmapscan_1_idx
                                 Index Cond: (material_bitmapscan_1.k = '2021-06-03 00:00:00'::timestamp without time zone)
         ->  Hash Join
               Output: share0_ref3.i
               Hash Cond: (share0_ref3.j = share0_ref2.j)
               ->  Result
                     Output: share0_ref3.i, share0_ref3.j
                     Filter: (share0_ref3.j = 2)
                     ->  Shared Scan (share slice:id 1:0)
                           Output: share0_ref3.i, share0_ref3.j
               ->  Hash
                     Output: share0_ref2.j
                     ->  Result
                           Output: share0_ref2.j
                           Filter: (share0_ref2.j = 2)
                           ->  Shared Scan (share slice:id 1:0)
                                 Output: share0_ref2.i, share0_ref2.j
 Optimizer: Pivotal Optimizer (GPORCA)

with mat as(
    select i, j from material_bitmapscan_1
    where i = 2 and j = 2 and k = timestamp '2021-06-03' and l = timestamp '2021-06-03'
)
select m1.i
from mat m1 join mat m2 on m1.j = m2.j;
 i 
---
(0 rows)
```

I might propose to remove a whole `AddBitmapFilterColumns()` call from `PdxlnBitmapTableScan()` one but not sure about negative side effects there are.